torch/nn/parallel/distributed.py

```diff
@@ -1305,3 +1305,27 @@ def _set_params_and_buffers_to_ignore_for_model(
         # during synchronization. It will be removed when the API is finalized
         # as part of addressing https://github.com/pytorch/pytorch/issues/43690.
         module._ddp_params_and_buffers_to_ignore = params_and_buffers_to_ignore
+
+
+    def get_ddp_logging_data(self):
+        r"""
+        This interface can be called after DistributedDataParallel() is
+        constructed. It returns DDPLoggingData for debugging and analysis.
+        More detailed explanation of the fields in DDPLoggingData are in
+        ``torch/c10/util/Logging.h``.
+        """
+        return self.logger._get_ddp_logging_data()
+
+    def set_ddp_runtime_logging_sample_rate(self, sample_rate):
+        r"""
+        This interface allows users to set sample_rate of collecting
+        runtime stats. The runtime stats will be recorded for the
+        first 10 iterations, after 10 iteratons runtime stats will be
+        recorded once every "sample_rate" training iterations. In
+        default, runtime stats are recorded for the first 10 iterations,
+        after 10 iterations runtime stats are recorded once every
+        "kDDPRuntimeLoggingSampleRate=100" training iterations.
+        """
+        if sample_rate <= 0:
```

Comment:
Should we check that the arg is an int and greater/equal to 1? Something like 0.5 for example doesn't make sense (and wouldn't that be cast down to 0 if it's passed into cpp as an int?)